### PR TITLE
No Bug - Hide loader only when loading is done.

### PR DESCRIPTION
**Pull request by adngdb**
QA tests rely on the loader gif element to disappear to start running their tests. As of this, that loader will get removed only when the dynamic_form library is done with everything it has to do. That will prevent weird race conditions to happen, and hopefully will make the tests more reliable.

### CODE REVIEW

File: webapp-django/crashstats/supersearch/static/supersearch/js/lib/dynamic_form.js [hunk 1]
```diff
@@ -124,6 +123,7 @@
                 if (onReadyCallback) {
                     onReadyCallback();
                 }
+                $('.loader', container).remove();
```
Comment (peterbe) [inline webapp-django/crashstats/supersearch/static/supersearch/js/lib/dynamic_form.js]: I don't like this. It just doesn't feel right. This function calls "onReadyCallback()" but it's clearly not ready if it does more. Just the name implies that we're lying :)

It also doesn't feel right to do more in a function after you've "exited" by calling the callback. 

What could happens is this:

```javascript
showLoader()
fetch(...).then(r => {
    someCallback(r)
    hideLoader()
})
function someCallback(data) {
    showTable(data)
}
```
Your DOM will then be...
```
<!-- frame 1 -->
<div class="loading">Please wait</div>
```
```
<!-- frame 2 -->
<div class="loading">Please wait</div>
<table>
  data
</table>
```
```
<!-- frame 3 -->
<table>
  data
</table>
```

**Comment by adngdb** [comment]
Alright, I move the removal of the loader to right before calling the callback. This would avoid that specific case you showed. 

Regarding using that loader for tests, it is, as far as selenium is concerned, the only sensible way to know that the form is ready to be used. And it is the correct thing to do, because that is also what users use to know they can interact with it. Also, this is a commit I made while talking to Matt about this very problem. :)

**Comment by peterbe** [comment]
Why can't the selenium tests instead depend on the form being **present**? 
```javascript
var wait = setInterval(function() {
    if ($('form.myclass:visible').length) {
      clearInterval(wait);
      proceedCallback();
   }
}, 500);
```

**Comment by peterbe** [comment]
Also, I don't think the cognitive process of the human is "the loader is gone, now I can type in what I want". I think it's more likely to be "the input field is there, now I can type in what I want". But that's honestly just a gut feeling. 

Anyway this is less important for the context of this PR. I'm just hoping @m8ttyB is joining in :)

**Comment by peterbe** [comment]
r+ by the way. I don't know if stage is "free".

**Comment by m8ttyB** [comment]
@peterbe The way that the tests are engineered is to look for an element on the page that signals that the dynamic elements in the page are loaded (enough) and the tests can begin to do their work. 

@adngdb and I met last Friday and focused in on using the loader as that keystone. I'm sure there are multiple routes to this end, but we focused in on this solution.

**Comment by peterbe** [comment]
Yeah, I'm not trying to block this. Just saying ... "have you considered ..."

**Comment by adngdb** [comment]
I'm actually closing this because I have integrated in #3478 as part of commit https://github.com/mozilla/socorro/pull/3478/commits/01798187eefe3d53580625c13e271cdb97e2f5e4.

**Pull request closed without merging**